### ISSUE TITLE
Bun integration

### ISSUE THREAD

**Issue by RajaARK99** [apollographql/apollo-server#7733]
Support bun in Apollo server.

**Comment by RajaARK99** [comment]
Use bun as a run time not using node.

> Can you expand? What is currently unsupported? Is [this issue](https://github.com/apollographql/apollo-server/issues/7369) related?

**Comment by trevor-scheer** [comment]
I think the issue I linked is related then. It requires Apollo Server to stop using Node APIs. Is that what's preventing AS from running in Bun or is there anything else?

A PR to fix https://github.com/apollographql/apollo-server/issues/7369 would be welcome. I won't be able to prioritize this work any time soon.

**Comment by michaelbromley** [comment]
Here's what I ran into when trying to get Apollo Server running on Bun:

- https://github.com/oven-sh/bun/issues/4947

**Comment by HuakunShen** [comment]
Query and Mutation works for me in bun v1.0.8, but subscription doesn't work. 
Here is an issue https://github.com/oven-sh/bun/issues/6905

Which NodeJS does apollo server use that bun doesn't?

**Comment by trevor-scheer** [comment]
As mentioned in the comment above and in the [duplicate issue here](https://github.com/apollographql/apollo-server/issues/7776), it would be nice to consider subscriptions as well when resolving this issue (though all subscriptions related work belongs to separate projects, so that would likely be a separate PR to another project).

**Comment by oliveredsberger** [comment]
Same issue as reported by @michaelbromley. @HuakunShen how did you get queries to work?

**Comment by HuakunShen** [comment]
@oliveredsberger It works out of the box.
Use this example
https://github.com/apollographql/docs-examples/tree/main/apollo-server/v4/getting-started

```bash
bun src/index.ts
```

I tried running TypeScript directly with bun, or run the compiled JS with bun. Both work.

**Comment by ChrisEnglert** [comment]
Hello, I've just made this:

https://github.com/ChrisEnglert/apollo-server-integration-bun

Allows using ApolloServer with Bun.serve({...}) API

maybe you'd like to consider this for @as-integrations ?

**Comment by trevor-scheer** [comment]
@ChrisEnglert that would be great! Would you like it to publish under the `@as-integrations` npm scope as well? Go ahead and transfer the repo to the `apollo-server-integrations` org if that's still something you'd like to do and we can go from there.

**Comment by ChrisEnglert** [comment]
@trevor-scheer yes, sure. I just need an invite I guess ("You don’t have the permission to create public repositories on apollo-server-integrations")
